### DISCUSSION TITLE
2.0.1 release

### DISCUSSION THREAD

**Discussion by eschutho** [apache/superset#21263] [General]
Hello, this is a call for requested PR cherries for the 2.0.1 release. Please list any PRs that are fixes for the 2.0.0 release. 

If you have any issues that haven't been fixed, you can paste them below, and a member of the community may or may not pick it up, but we'll have a better chance of getting the fix in the release if there is already a PR associated with it. 

Please submit any cherries to this discussion no later than September 14. After that period, the release will be considered closed and new cherries should go to the next patch release. 

Thank you!

Here is the [current list](https://github.com/apache/superset/issues?q=label%3Av2.0.1) of cherries: 

Didn't make it into 2.0.1
https://github.com/apache/superset/pull/21208 - removed because it didn't merge cleanly. It's based off a new feature.
https://github.com/apache/superset/pull/21070 - removed because this is a feature
https://github.com/apache/superset/pull/21489 - removed because it didn't merge cleanly. It's based off newer feature: [21163](https://github.com/apache/superset/pull/21163)
https://github.com/apache/superset/pull/21450 - removed because it didn't merge cleanly. It's based off newer code.
https://github.com/apache/superset/pull/21484 - removed because it didn't merge cleanly. It's based off newer feature: [21163](https://github.com/apache/superset/pull/21163)
https://github.com/apache/superset/pull/20872 - removed because PR needs to be reviewed/still open
https://github.com/apache/superset/pull/21161 - removed because it didn't merge cleanly. It is based of sqlalchemy 1.4
https://github.com/apache/superset/pull/20081 - removed because it didn't merge cleanly. It is based of sqlalchemy 1.4
https://github.com/apache/superset/pull/21114 - removed because it didn't merge cleanly. It's based off a new feature.
https://github.com/apache/superset/pull/18063 - removed because PR is still open
https://github.com/apache/superset/pull/21920 - was based off newer code.
https://github.com/apache/superset/pull/21869 - came in after the cutoff

Open bugs: 
https://github.com/apache/superset/issues/20576 - fix is a feature with migrations - can't merge
https://github.com/apache/superset/issues/16658 - not a regression for 2.0 but if there's a fix, I'll cherry it
https://github.com/apache/superset/issues/20961 - fix is based of newer code (21450)
https://github.com/apache/superset/issues/19940
https://github.com/apache/superset/issues/19988

**Comment by justin-tomlinson** [comment]
Putting forward these as per recent slack convo: 
[20819](https://github.com/apache/superset/pull/20819), [21084](https://github.com/apache/superset/pull/21084), [21208](https://github.com/apache/superset/pull/21208)

**Reply by justin-tomlinson** [reply]
also this isn't merged yet but has been approved and would be useful [21070](https://github.com/apache/superset/pull/21070)

**Reply by eschutho** [reply]
Thank you @justin-tomlinson!

**Comment by kamalkeshavani-aiinside** [comment]
List of PRs, requesting to add:
https://github.com/apache/superset/pull/21216, https://github.com/apache/superset/pull/21154, https://github.com/apache/superset/pull/21153, https://github.com/apache/superset/pull/21114, https://github.com/apache/superset/pull/21076, https://github.com/apache/superset/pull/21044, https://github.com/apache/superset/pull/21032, https://github.com/apache/superset/pull/21007, https://github.com/apache/superset/pull/21005, https://github.com/apache/superset/pull/20969, https://github.com/apache/superset/pull/20962, https://github.com/apache/superset/pull/20946

List of Bugs, hoping to be fixed:
https://github.com/apache/superset/issues/20930 (Major bug/downgrade in 2.0)
https://github.com/apache/superset/issues/20576
https://github.com/apache/superset/issues/16658

**Reply by eschutho** [reply]
Thank you @kamalkeshavani-aiinside. I've added all these cherries to the release. @eric-briscoe is currently reviewing the open bugs and seeing if there's been a fix yet.

**Reply by kamalkeshavani-aiinside** [reply]
Adding one more https://github.com/apache/superset/pull/21294

**Reply by kamalkeshavani-aiinside** [reply]
@eschutho In case we go for RC2, please add #21729.

**Comment by sfirke** [comment]
The merged PRs above are great.  In terms of unsolved bugs, +1 to #20930.  I've been vexed by #20961, no idea how quick a fix that would require.

**Reply by sfirke** [reply]
I see #20961 just got closed by https://github.com/apache/superset/pull/21450, could #21450 possibly be included in 2.0.1 ?  Sorry to request it late.

**Reply by eschutho** [reply]
Added, thanks!

**Comment by cccs-tom** [comment]
Could we include the permissions fix from #20382, please?

**Reply by eschutho** [reply]
added 👍

**Comment by mdeshmu** [comment]
Please add [20733](https://github.com/apache/superset/pull/20733) and [20736](https://github.com/apache/superset/pull/20736) together.

**Reply by eschutho** [reply]
added 👍

**Comment by nigzak** [comment]
I suggest to update the container runtime itself, I get so many warnings (open CVEs) in cloud if I fetch the superset 2.0.0 container (similar to 1.x.x) - this should also be in general updated

Source: Dockerhub container 2.0.0 https://hub.docker.com/r/apache/superset


my dockerfile update script to show what is currently open (last checked 2 weeks ago, could now even more)
```
# update security stuff in superset 2.0.0
RUN apt-get update
RUN apt-get install curl -y
RUN apt-get install libcurl3-gnutls -y
RUN apt-get install libcurl4  -y
RUN apt-get install libcurl4-openssl-dev -y
RUN apt-get install dpkg -y
RUN apt-get install gzip -y
RUN apt-get install xz-utils -y
RUN apt-get install libexpat1 -y
RUN apt-get install libexpat1-dev -y
RUN apt-get install libtirpc-common -y
RUN apt-get install libtirpc-dev -y
RUN apt-get install libtirpc3 -y
RUN apt-get install libxml2  -y
RUN apt-get install libxml2-dev -y
RUN apt-get install linux-libc-dev -y
RUN apt-get install openssl -y
RUN apt-get install zlib1g -y
RUN apt-get install zlib1g-dev -y
RUN apt-get install subversion -y
```

**Reply by nigzak** [reply]
its now even more issues in superset 2.0.0 docker image which are open, I had to add this fixes too

RUN apt-get install unzip -y
RUN apt-get install libxslt1-dev -y
RUN apt-get install libxslt1.1 -y
RUN apt-get install libgdk-pixbuf2.0-bin -y
RUN apt-get install libgdk-pixbuf-2.0-dev -y
RUN apt-get install libgdk-pixbuf2.0-common -y
RUN apt-get install libgdk-pixbuf-2.0-0 -y
RUN apt-get install gir1.2-gdkpixbuf-2.0 -y

**Comment by dpgaspar** [comment]
please add https://github.com/apache/superset/pull/21272

**Reply by eschutho** [reply]
added 👍

**Comment by D3nn3** [comment]
Fixing the following issue for async queries would also be nice #19940 . But I assume it's not so easy and probably won't make it in 2.0.1.

**Comment by villebro** [comment]
I'd like to see this one added: #21296 (I actually already tagged it with 2.0 😁)

**Reply by eschutho** [reply]
👍

**Comment by zedlauzy** [comment]
SQLAlchemy should be upgraded to 1.4: #19890

**Reply by eschutho** [reply]
@betodealmeida what do you think about this PR for a patch release?

**Reply by betodealmeida** [reply]
Sounds risky to me!

**Reply by daniel-sali** [reply]
Do you happen to have a planned date / version for the migration to SQLAlchemy 1.4?

**Reply by eschutho** [reply]
it's in the 2.1 release which is currently being voted on.

**Comment by mdeshmu** [comment]
Please include [21161](https://github.com/apache/superset/pull/21161) and [20081](https://github.com/apache/superset/pull/20081)

**Reply by eschutho** [reply]
added. Thank you @mdeshmu!

**Comment by mdeshmu** [comment]
Please include:

[21380](https://github.com/apache/superset/pull/21380)
[21419](https://github.com/apache/superset/pull/21419)
[20493](https://github.com/apache/superset/pull/20493)
[20714](https://github.com/apache/superset/pull/20714)
[20178](https://github.com/apache/superset/pull/20178)
[20872](https://github.com/apache/superset/pull/20872)
[21091](https://github.com/apache/superset/pull/21091)

**Reply by eschutho** [reply]
added! Thanks!

**Comment by dlabmx** [comment]
Please include #20748 / #20797

**Reply by eschutho** [reply]
Thanks @dlabmx! I don't see a fix for this issue yet, but I'll keep an eye out for a PR.

**Reply by ryerraguntla** [reply]
@dlabmx I will be interested with fix. Looking forward for the PR

**Reply by sfirke** [reply]
Looks like the in-review PR https://github.com/apache/superset/pull/21576 will fix these as well as #20930 listed in "Open Bugs" in the top post

**Reply by sfirke** [reply]
Oh nice I see it got merged 👀

**Comment by nigzak** [comment]
!!!!!!!!!!!!!
UDPATE: +++ IGNORE THIS COMMENT/CHANGE REQUIEST +++
!!!!!!!!!!!!!

Found an issue in the deployment configuration - the feature flags were configured TWICE, one time with the RDBAC flag, one time without it ... removing the wrong duplicated configuration solved the issue in superset 2.0.0


/// HISTORY - IGNORE THIS, THIS IS NOT CORRECT ///
you should include a fix for the feature flag DASHBOARD_RBAC  bug #19988 ... currently it does not work in superset 2.0.0 also.
I have activated it but the field is not visible so it's a very painful issue for access control ...

seems bug came with Superset 1.5.0 and is still not fixed

**Reply by nigzak** [reply]
seems also there is another bug ticket for same issue with 1.4.x  https://github.com/apache/superset/issues/18634

**Comment by D3nn3** [comment]
Would love to see the recent fixes #21484 and #21489

**Reply by eschutho** [reply]
added, thanks!

**Comment by mdeshmu** [comment]
Can [21326](https://github.com/apache/superset/pull/21326) be included too, please.

**Reply by eschutho** [reply]
This looks more like an improvement rather than a fix, so I'm going to recommend that this PR goes into the 2.1 release.

**Comment by mathieudruart** [comment]
Hi, we are interested in adding these PR :

#20061 
#20063
#21498

Thank you

**Reply by eschutho** [reply]
added!

**Comment by sfirke** [comment]
I see someone noting in the bug thread at https://github.com/apache/superset/issues/20845 that it is fixed by cherry-picking #20651, any chance that cherry could be added too?

**Reply by zhaorui2022** [reply]
Yes, we did a cherry pick and after that, the broken dashboards affected by the bug were all fixed.

**Reply by eschutho** [reply]
I'm about to build the release, but if this merges in cleanly, I'll give it a shot.

**Comment by mdeshmu** [comment]
can we include 

[21549](https://github.com/apache/superset/pull/21549)
[21561](https://github.com/apache/superset/pull/21561)

**Reply by eschutho** [reply]
I'm about to build the release, but if these merge in cleanly, I'll give it a shot.

**Reply by eschutho** [reply]
There's a failing test on the second PR, so it will need to be addressed before it can get merged.

**Reply by mdeshmu** [reply]
[21561](https://github.com/apache/superset/pull/21561) is merged.

**Reply by eschutho** [reply]
added

**Comment by zhaorui2022** [comment]
Can we also include https://github.com/apache/superset/pull/21561 as it was blocking our users to use the new "set up email report" feature from charts and reports?

**Reply by eschutho** [reply]
same comments as above regarding the failing test.

**Reply by eschutho** [reply]
added

**Comment by zhaorui2022** [comment]
Is it too late to include #20830? This is a small fix but without this, when it times out to take screenshots for alerts & reports, it can result in many different kinds of error messages that are confusing.

**Reply by eschutho** [reply]
included

**Comment by zhaorui2022** [comment]
We have seen a regression after upgraded to 2.0.0 related with SELLAB_BACKEND_PERSISTENCE set to True by default. #21665 I am not sure if others have seen the same issue. If others also experience the same, should we set this feature flag to False to avoid breaking experience? We have the feature flag enabled for 2 days, and after disabling it later causes another bug where users lose queries tabs history in SQL editor. So it might be a good idea if it's not enabled by default until the bug is resolved.

**Reply by eschutho** [reply]
Thanks for flagging. We can try to look into this issue before 2.0.2

**Reply by eschutho** [reply]
I tested this in 2.0.1 and it seems to be fixed there.

**Comment by dbeaudev** [comment]
Can we please include the following issues? #21576  #20797     From what I can tell, 21576 looks like it also includes the fix for 20797.  Both are very important issues for us (and from slack channel, looks like many others are also running into it).  Thanks!

**Reply by eschutho** [reply]
we've already finished the cherrying, but we can try to get these into 2.0.2

**Reply by kamalkeshavani-aiinside** [reply]
@eschutho #21576 is a very important fix, which solves a basic problem reported by many users. Please consider to include it in 2.0.1.

**Reply by eschutho** [reply]
@kamalkeshavani-aiinside I pushed the rc for vote up today, but if there are any changes required, there's still a chance to sneak more cherries in. I'll see what I can do. If not, we'll be aiming to fast follow with a 2.0.2.

**Reply by kamalkeshavani-aiinside** [reply]
@eschutho Were you able to include #21576 in 2.0.1 rc2?

**Reply by eschutho** [reply]
@kamalkeshavani-aiinside yes, I was.

**Reply by kamalkeshavani-aiinside** [reply]
@eschutho Thank you for confirmation. I have mentioned in the PR as well, but I am able to reproduce some tab related issues even with this fix in rc2. Hoping to get a new fix soon.
BTW any news on rc3?

**Reply by eric-briscoe** [reply]
@kamalkeshavani-aiinside @sfirke @dbeaudev  After some complex investigation @eschutho and @AAfghahi  and I have realized #21576 is not a viable fix for 2.0.1.  The issues were fixed and work in master, but the fix was built upon a large feature PR that is not eligible for the 2.0.1 patch release: https://github.com/apache/superset/pull/19983/files#diff-af1772685d6f051784b8b0360a2ce252d6b39242970ca25f1ada88e0deb32e80
Because #21576 depends on the feature work the cherry-pick of #21576 does not fix the issue in 2.0.1 and actually introduces some incompatible code.  After digging through this for a number of hours we have determined the best scenario is to back out #21576 from 2.0.1 as it does not actually fix anything on the 2.0.1 baseline and may actually cause other unintended behaviors.  The fix for https://github.com/apache/superset/pull/21576 https://github.com/apache/superset/issues/20797 done in #21576 will be included in 2.1.0

**Reply by sfirke** [reply]
Thanks you three for digging into it and @eric-briscoe for the clear explanation!  Much appreciated.

**Reply by adimyth** [reply]
@eric-briscoe Has the fix for #20797 in #21576 merged & released? When can we expect it to be released?

**Reply by sfirke** [reply]
@adimyth my understanding is that it will definitely be in 2.1.0 as 2.1.0 will be cut from the master branch.  Agreed it's still a painful issue, I have users avoiding live demos using Superset because of it and upper management brought it up in a meeting yesterday ("does Superset still have that bug where the dashboards don't load?")

**Reply by dbeaudev** [reply]
@eric-briscoe Is it possible to get update on when 2.1.0 will be released.  This is very problematic for us as well.  So much that we may need to start looking for other options.  I hate to go that route when it has already been fixed.  Yet, we have been waiting months now to get it into production.

**Reply by adimyth** [reply]
@eric-briscoe @sfirke can you guys provide an update on when v2.1.0 will be released? We are unable to ship superset to production because of this issue. We wouldn't all our efforts to go down the drain, just because of this one issue.

**Reply by eschutho** [reply]
2.1.0 is currently in voting and should be released soon.

**Comment by mattitoo** [comment]
I posted https://github.com/apache/superset/issues/21731 as a Bug for RC1, don't know if I did it correctly. :)

**Reply by eschutho** [reply]
Thank you. This is perfect. We'll take a look at this.

**Comment by rizacavus** [comment]
Can we please include the following PRs?
#20801, #21566

**Reply by eschutho** [reply]
we can only add bug fixes to 2.0.1, but this will go into the 2.1 release as a feature.

**Comment by NephilimJaeger** [comment]
Can you please include this fix?  #18063

**Reply by eschutho** [reply]
It is still open and has some merge conflicts that I just resolved. I'll see if tests pass and we can merge it in.

**Comment by eschutho** [comment]
Thank you all for flagging PRs and bugs for 2.0.1! 🙏  Our goal is to make this and future patch releases for the 2.0 branch stable and bug-free. Rc1 is currently up for vote, so please test and vote on this release. We will continue to push our more patch releases and add in as many fixes as we can.

**Comment by eschutho** [comment]
From slack: https://github.com/apache/superset/pull/21869

**Comment by eschutho** [comment]
From slack: https://github.com/apache/superset/pull/21920

**Comment by cbuffevant** [comment]
Can we please include this one https://github.com/apache/superset/pull/22280  in rc5?

**Comment by sfirke** [comment]
### Testing 2.0.1rc5

Here are the PRs I was able to test from 2.0.1rc5 and my notes on them, in case others want to corroborate mine or test different ones:
- ✅  #21296 zeroes now show in tooltip
- ✅  #20819, ALSO 21294 GENERIC_CHART_AXES with label no longer causes x-axis time variable to appear as numeric format.  I don't remember which chart type I checked this on, only saw one PR at first and now realize this was two PRs patching the same behavior on different chart types (?)
- ✅, 😕   #21007 Gauge Chart now doesn't get cut off when resized.  It technically has a tooltip but it does not display series name:
![image](https://user-images.githubusercontent.com/7569808/207649892-1249b7ca-fa1d-4a83-b25c-c3e22891bf62.png)
- ❌ Handlebars chart is now broken, in what looks like the same way as on master.  Could be #21005 as that's the only handlebars-specific PR I see?  The sorting behavior that 21005 addressed indeed appears fixed.

Here's the broken handlebars chart:
![image](https://user-images.githubusercontent.com/7569808/207703094-01e4ab33-090e-4039-914f-1aeba9a30420.png)

I don't think the lack of series name on #21007 tooltip it should hold up the release, just get filed as a new bug.  I didn't test features for that chart like multiple series and label overlap.

**Reply by rusackas** [reply]
The issue being reported with Handlebars is due to a change in Superset’s configuration. In the current default configuration of Superset, it’s only possible to use the Handlebars chart with limited HTML tags (similar to Github’s markdown) and no CSS styling. This is because the Handlebars plugin utilizes the `<style>` tag, which is disabled by default. Additionaly, people use the `style` and `class` attributes in the tags they utilize in their templating — these are also currently disabled by default. To enable the Handlebars plugin with all of the above CSS capabilities, you can edit your `config.py` file to set `HTML_SANITIZATION_SCHEMA_EXTENSIONS` with some additional config overrides as follow:
```js 
HTML_SANITIZATION_SCHEMA_EXTENSIONS = {
  "attributes": {
    "*": ["style","className"],
  },
  "tagNames": ["style"]
}
```
Please note that if you wish to do so, it is strongly advised to enable a content security policy (CSP) that only allows content to be loaded from trusted domains. More details for these changes and the risks involved will be forthcoming after the 2.0.1 release is made official. Also note that a there was an issue in RC5 where the `HTML_SANITIZATION_SCHEMA_EXTENSIONS` configuration was not being picked up properly by the Handlebars plugin, but that is resolved in RC6.
